From c3b0c7ca98a15ae665cf206e19425e62f654dffa Mon Sep 17 00:00:00 2001
From: Iceows <mounierr07@gmail.com>
Date: Mon, 17 Nov 2025 19:54:23 +0100
Subject: [PATCH 4/7] device: fix configstore crash, secomp_policy configstore
 don't have getpid and gettid

Fix configstore crash on A15 - vendor configstore don't have getpid and gettid

Co-Authored-By: Pierre-Hugues HUSSON <phh@phh.me>
---
 rw-system.sh | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/rw-system.sh b/rw-system.sh
index ac31458..be24c49 100644
--- a/rw-system.sh
+++ b/rw-system.sh
@@ -380,8 +380,10 @@ mkdir -p /mnt/phh/
 mount -t tmpfs -o rw,nodev,relatime,mode=755,gid=0 none /mnt/phh || true
 mkdir /mnt/phh/empty_dir
 touch /mnt/phh/empty
-fixSPL
+touch /mnt/phh/unreadable
+chmod 0 /mnt/phh/unreadable
 
+fixSPL
 changeKeylayout
 
 mount /mnt/phh/empty /vendor/bin/vendor.samsung.security.proca@1.0-service || true
@@ -641,6 +643,11 @@ if getprop ro.vendor.build.fingerprint | grep -iq -e Redmi/rosemary \
     mount -o bind /system/phh/rosemary-excluded-input-devices.xml /system/etc/excluded-input-devices.xml
 fi
 
+# Fix configstore crash on A15 - vendor configstore don't have getpid and gettid
+if [ "$vndk" -le 29 ]; then
+    mount /mnt/phh/unreadable /vendor/etc/seccomp_policy/configstore@1.1.policy
+fi
+
 if getprop ro.vendor.build.fingerprint | grep -iq -E -e 'huawei|honor' || getprop persist.sys.overlay.huawei | grep -iq -E -e 'true'; then
     p=/product/etc/nfc/libnfc_nxp_*_*.conf
     mount -o bind "$p" /system/etc/libnfc-nxp.conf ||
-- 
2.34.1

